src/ResourceManager/DataFactories/Commands.DataFactoryV2/DataFactories/UpdateAzureDataFactoryCommand.cs

```diff
@@ -0,0 +1,94 @@
+        [Alias(Constants.DataFactoryName)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryObject, Position = 0, Mandatory = true, ValueFromPipeline = true,
+            HelpMessage = Constants.HelpFactoryObject)]
+        [ValidateNotNullOrEmpty]
+        public PSDataFactory InputObject { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ByResourceId, Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpResourceId)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryName, Position = 2, Mandatory = false,
+            HelpMessage = Constants.HelpTagsForFactory)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryObject, Position = 1, Mandatory = false,
+            HelpMessage = Constants.HelpTagsForFactory)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByResourceId, Position = 1, Mandatory = false,
+            HelpMessage = Constants.HelpTagsForFactory)]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryName, Position = 2, Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpIdentityForFactory)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryObject, Position = 1, Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpIdentityForFactory)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByResourceId, Position = 1, Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpIdentityForFactory)]
+        public FactoryIdentity Identity { get; set; }
+
+        [EnvironmentPermission(SecurityAction.Demand, Unrestricted = true)]
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName.Equals(ParameterSetNames.ByFactoryObject, StringComparison.OrdinalIgnoreCase))
+            {
+                Name = InputObject.DataFactoryName;
+                ResourceGroupName = InputObject.ResourceGroupName;
+            }
+            else if (ParameterSetName.Equals(ParameterSetNames.ByResourceId, StringComparison.OrdinalIgnoreCase))
+            {
+                var parsedResourceId = new ResourceIdentifier(ResourceId);
+                Name = parsedResourceId.ResourceName;
+                ResourceGroupName = parsedResourceId.ResourceGroupName;
+            }
+
+            var parameters = new UpdatePSDataFactoryParameters()
+            {
+                ResourceGroupName = ResourceGroupName,
+                DataFactoryName = Name,
+                Tags = Tag,
+                Identity = Identity
+            };
+
+            WriteObject(DataFactoryClient.UpdatePSDataFactory(parameters));
```

Comment:
@hvermis looks like there is no `ShouldProcess` block wrapping this call

Reply:
Done